File: src/lessons/bat/string1/StringYakEntity.java
package lessons.bat.string1;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class StringYakEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(stringYak((String)param[0])));
    }
  }

  /* BEGIN TEMPLATE */
  String stringYak(String str)
  {
    /* BEGIN SOLUTION */
    String result = "";

    for (int i = 0; i < str.length(); i++) {
      // Look for i starting a "yak" -- advance i in that case
      if (i + 2 < str.length() && str.charAt(i) == 'y' && str.charAt(i + 2) == 'k') {
        i = i + 2;
      } else { // Otherwise do the normal append
        result = result + str.charAt(i);
      }
    }

    return result;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/bat/string1/StringYakEntity.py
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(stringYak(param[0])))

# BEGIN TEMPLATE
def stringYak(str):
# BEGIN SOLUTION
  res = ''
  i=0
  while i<len(str):
    if i+2<len(str)  and str[i] == 'y' and str[i+2]=='k':
      i += 2
    else:
      res += str[i]
    i+=1
  return res
# END SOLUTION
# END TEMPLATE
